inlang/source-code/paraglide/paraglide-js/src/compilerV2/compileExpression.ts

```diff
@@ -0,0 +1,51 @@
+import type { Expression, FunctionAnnotation } from "@inlang/sdk2"
+import { isValidJSIdentifier } from "../services/valid-js-identifier/index.js"
+import {
+	escapeForDoubleQuoteString,
+	escapeForSingleQuoteString,
+} from "../services/codegen/escape.js"
+import { mergeTypeRestrictions, type Compilation } from "./types.js"
+
+export function compileExpression(lang: string, expression: Expression): Compilation<Expression> {
+	if (expression.annotation) {
+		const fn = expression.annotation
+		const hasOptions = fn.options.length > 0
+
+		let typeRestrictions: Record<string, string> = {}
+		if (fn.name === "plural" && expression.arg.type === "variable") {
+			typeRestrictions[expression.arg.name] = "number"
+		}
+
+		const args = [`"${lang}"`, compileArg(expression.arg)]
+		if (hasOptions) {
+			const options = compileOptions(fn.options)
+			args.push(options.code)
+			typeRestrictions = mergeTypeRestrictions(typeRestrictions, options.typeRestrictions)
+		}
+
+		const code = `registry.${fn.name}(${args.join(", ")})`
+		return { code, typeRestrictions, source: expression }
+	}
+	const code = compileArg(expression.arg)
+	return { code, typeRestrictions: {}, source: expression }
+}
+
+function compileOptions(
+	options: FunctionAnnotation["options"]
+): Compilation<FunctionAnnotation["options"]> {
+	const entires: string[] = options.map((option) => `${option.name}: ${compileArg(option.value)}`)
+	const code = "{" + entires.join(", ") + "}"
+	// TODO Type-Narrowing for options
+	return { code, typeRestrictions: {}, source: options }
+}
+
+function compileArg(arg: Expression["arg"]): string {
+	switch (arg.type) {
+		case "literal":
+			return `"${escapeForDoubleQuoteString(arg.name)}"`
+		case "variable": {
+			const escaped = !isValidJSIdentifier(arg.name)
```

Comment:
Aha. Function, variable, input names can't be compiled if they are not valid JS identifiers. Should we enforce `snake_case` for Names in the inlang SDK? 

- `snake_case` is the most interoperable format 
- i fear that we will get into `my-cool-function` scenarios which can't be compiled in several programming languages, or `myCoolFunction` which requires mapping in other programming languages that are case insensitive (does that exist?)

Reply:
We could have a validation rule for it though "use_snake_case_validation_rule"

Reply:
Paraglide itself could work with any arbitrary string for:
- IDs & Aliases
- Registry Function names & arguments
- Message Inputs

From Paraglide's perspective we don't need to enforce any specific formatting. I don't know about other Inlang apps though

Reply:
Interesting. Then let's skip the enforcement